### PR TITLE
Refactor new appointment flow tests

### PR DESCRIPTION
**Pull request by jbalboni**
## Description
This PR looks bigger than it is. What I have done:

1. Move type of care test block to the start of the file, to better follow the order in the newAppointmentFlow.js file
2. Grouped tests for each page into "next page" and "previous page" blocks
3. Added a few new tests to cover missed cases
4. Removed any tests that were not testing next/previous functions, because we're just rewriting config data in a test.

## Testing done
Unit tests?

## Acceptance criteria
- [ ] Tests pass

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/vaos/newAppointmentFlow.js [hunk 1]
```diff
@@ -63,13 +63,6 @@ export default {
   home: {
     url: '/',
   },
-  typeOfAppointment: {
```
Comment (jbalboni) [inline src/applications/vaos/newAppointmentFlow.js]: Just going to remove this until we actually do a provider flow

File: src/applications/vaos/newAppointmentFlow.js [hunk 2]
```diff
@@ -167,15 +167,8 @@ export default {
       if (eligibilityStatus.direct) {
         let appointments = null;
 
-        // If we can't get the history, then continue anyway
-        // and we'll show the full clinic list
```
Comment (jbalboni) [inline src/applications/vaos/newAppointmentFlow.js]: I think this is actually incorrect, we should go to the request flow if we can't get appt history.

File: src/applications/vaos/newAppointmentFlow.js [hunk 3]
```diff
@@ -190,9 +183,8 @@ export default {
             dispatch(startDirectScheduleFlow(appointments));
             return 'clinicChoice';
           }
-        } else {
-          dispatch(startDirectScheduleFlow(appointments));
-          return 'clinicChoice';
+        } catch (error) {
+          captureError(error);
```
Comment (narin) [inline src/applications/vaos/newAppointmentFlow.js]: If we can't get appointment history, should we `dispatch(startRequestAppointmentFlow())` in the catch block to send them to the request flow?

**Review by narin — APPROVED** [review]
1 comment, but otherwise the refactor of the tests LGTM. Order makes sense and is more readable with the next/prev blocks.

File: src/applications/vaos/newAppointmentFlow.js (hunk 3 above)
Comment (jbalboni) [reply src/applications/vaos/newAppointmentFlow.js]: Yeah, if you expand the file below this change you can see that if they fall through here we'll check that they're eligible for requests and start that flow.

**Review by bkjohnson — APPROVED** [review]

**Pull request merged**